From 48aba98c8cf0ab0bda030e93b78e95c56c254787 Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Fri, 28 Mar 2014 17:10:11 +0000
Subject: libparted: copy flags when duplicating GPT partitions

* libparted/labels/gpt.c (gpt_partition_duplicate): Copy flags to new
partition.

Reported by Chris Murphy in https://bugzilla.redhat.com/747497.

Origin: upstream, http://git.debian.org/?p=parted/parted.git;a=commitdiff;h=febeedd8d3d
Forwarded: not-needed
Last-Update: 2011-05-15

Patch-Name: libparted-gpt-copy-flags.patch
---
 libparted/labels/gpt.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/libparted/labels/gpt.c b/libparted/labels/gpt.c
index 68102a3..03154d2 100644
--- a/libparted/labels/gpt.c
+++ b/libparted/labels/gpt.c
@@ -1743,9 +1743,7 @@ gpt_partition_duplicate (const PedPartition *part)
   if (!result_data)
     goto error_free_part;
 
-  result_data->type = part_data->type;
-  result_data->uuid = part_data->uuid;
-  strcpy (result_data->name, part_data->name);
+  *result_data = *part_data;
   return result;
 
 error_free_part:
